Updated Makefile and Documentation

benglines opened the pull request:
I changed the Makefile to require the user to specify the part name rather than expecting the user to only use the xc7a35tcsg324-1 part or to go into the Makefile to change the part name.

I also changed the Makefile to generate an .html page with a "frame view" of a bitstream. The feature was already implemented, but the Makefile made no use of it. Now it will be generated by default.

I updated the documentation to be more clear of the different features included in this tool. I also updated it to reflect changes in the Makefile that require the user to specify the part name. I also Changed the README from Markdown to the reStructuredText format that seems to be used in the majority of SymbiFlow repos.

benglines committed:
Removed Predefined Part in Makefile

The make command will now require the user to specify the PART variable

Signed-off-by: Ben Glines <benglines1@gmail.com>

benglines committed:
Changed Makefile to create "Frame View"

By default, the Makefile will now generate a .html file for the "Frame View"

Signed-off-by: Ben Glines <benglines1@gmail.com>

benglines committed:
Update README file

Changed to reStructuredText format. Updated to reflect other changes and to be more clear

Signed-off-by: Ben Glines <benglines1@gmail.com>

mithro commented:
@kgugala Can you find someone to review this?

ldalek-antmicro commented on Makefile:
```diff
@@ -37,9 +37,6 @@ clean::
 
-PART = xc7a35tcsg324-1
-#PART = xc7a50tfgg484-1
-
```
I would rather keep default part in makefile. You can remove commented-out line and move default one to the top of the file and then change assign operator to `?=` like this: `PART ?= xc7a35tcsg324-1`
(reference https://www.gnu.org/software/make/manual/html_node/Setting.html)

benglines replied:
Okay! I put in a line for the default part as you suggested.

ldalek-antmicro commented on Makefile:
```diff
@@ -75,5 +72,7 @@ bitstreamData.json: bitstream.bits bithtml.py grid/.dir frames/.dir
 		--dump-grid=$@ --grid-dir=$$PWD/grid
 
+all:: bitstream.html
+
```
I can't see any point in generating `bitstream.html` by default. It's not used anymore and it was left just for reference. In fact it should be removed (with the corresponding code in `bithtml.py`) or cleaned and left as optional target.

benglines replied:
I changed the file to not have bitstream.html build by default

ldalek-antmicro commented on README.rst:
```diff
@@ -0,0 +1,24 @@
+Bitstream Viewer
+================
+
+This tool has two primary functions:
```
Please left the original info from README.md:
`The purpose of this tool is to visualize the bits and their corresponding tags in a given bitstream.`

benglines replied:
I put that description back into the file

ldalek-antmicro commented on README.rst:
```diff
@@ -0,0 +1,24 @@
+
+This tool has two primary functions:
+
+1. Tile View: A grid of tiles will be generated to display the features in a bitstream. Each tile that is inhabited by features will be highlighted. By clicking on a highlighted tile, a pop-up will appear with the features within that tile and their corresponding bits. This is accomplished with `Vue.js, <https://vuejs.org/>`_ a JavaScript framework. 
```
Like this description :+1:

ldalek-antmicro commented on README.rst:
```diff
@@ -0,0 +1,24 @@
+
+1. Tile View: A grid of tiles will be generated to display the features in a bitstream. Each tile that is inhabited by features will be highlighted. By clicking on a highlighted tile, a pop-up will appear with the features within that tile and their corresponding bits. This is accomplished with `Vue.js, <https://vuejs.org/>`_ a JavaScript framework. 
+
+2. Frame View: A grid of frames will be generated to display features in a bitstream. Each frame that contains bits associated with a feature used by the bitstream will be highlighted. By clicking on a highlighted frame, the page will be redirected to a grid of all the bits for that specific frame.  
```
Point that this is an optional function

benglines replied:
I added a line to show that it is an optional function

benglines committed:
Update Makefile

bitstream.html will not be built by default, and the default part line was put back in place

Signed-off-by: Ben Glines <benglines1@gmail.com>

benglines committed:
Update README.rst

Frame view is now specified as an optional feature

Signed-off-by: Ben Glines <benglines1@gmail.com>

ldalek-antmicro reviewed (approved):
Looks good. One minor thing. Can you sign your commits (-s)?

ldalek-antmicro reviewed (changes requested):
Looks good. One minor thing. Can you sign your commits (-s)?

ldalek-antmicro commented:
@benglines please make sure to signoff your commits as without CI fails at the DCO check. Once you fix it we can merge the PR.

benglines commented:
Sorry I did not do this earlier. I believe everything should be in order
now. Let me know if it's still not right

On Tue, Jan 5, 2021 at 3:41 AM Lukasz Dalek <notifications@github.com>
wrote:

> @benglines <https://github.com/benglines> please make sure to signoff
> your commits as without CI fails at the DCO check. Once you fix it we can
> merge the PR.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/SymbiFlow/symbiflow-bitstream-viewer/pull/10#issuecomment-754556551>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/APKLMY23LXHCRCBVUU3IRBDSYLUG7ANCNFSM4PL7KXRA>
> .
>

ldalek-antmicro commented:
@benglines Can add empty line between commit description and Signed-off-by in https://github.com/SymbiFlow/symbiflow-bitstream-viewer/pull/10/commits/54f856c0fb7d531f0c38c6195bbb96ad18f1fd3e to satisfy DCO? Thanks!

benglines commented:
Done!

On Thu, Jan 7, 2021 at 6:12 AM Lukasz Dalek <notifications@github.com>
wrote:

> @benglines <https://github.com/benglines> Can add empty line between
> commit description and Signed-off-by in 54f856c
> <https://github.com/SymbiFlow/symbiflow-bitstream-viewer/commit/54f856c0fb7d531f0c38c6195bbb96ad18f1fd3e>
> to satisfy DCO? Thanks!
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/SymbiFlow/symbiflow-bitstream-viewer/pull/10#issuecomment-756107971>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/APKLMY3OYGLQTMRVEYJWN3TSYWXM7ANCNFSM4PL7KXRA>
> .
>

ldalek-antmicro reviewed (approved):
Thanks!
